src/plugins/intel_gpu/src/graph/program.cpp

```diff
@@ -1260,6 +1260,10 @@ void program::dump_program(const char* stage,
                            bool with_full_info,
                            std::function<bool(program_node const&)> const& filter) const {
     std::string path = get_dir_path(_config);
+    GPU_DEBUG_GET_INSTANCE(debug_inst);
+    GPU_DEBUG_IF(debug_inst->dump_graphs.size()) {
+        path = debug_inst->dump_graphs;
+    }
```

Comment:
can we make it happen with `ExecutionConfig::apply_debug_options` function? It has other config, too.

Reply:
Used apply_user_properties() because apply_debug_options() is protected function.